feat: validate --host flag for IP address

abuchanan-airbyte opened the pull request:
<img width="782" alt="image" src="https://github.com/user-attachments/assets/1709b63b-0d0e-4baf-a217-99f2afcbf056">

Since this validation happens before the `telemetry.Wrap` call, this should eliminate this category of errors from our error tracking.

colesnodgrass commented on internal/cmd/local/local_install.go:
```diff
@@ -39,6 +41,17 @@ func (i *InstallCmd) Run(ctx context.Context, provider k8s.Provider, telClient t
+	}
+
+	for _, host := range i.Host {
+		if regexp.MustCompile(`\d+\.\d+\.\d+\.\d+`).MatchString(host) {
```
We should replace this with the [net.ParseIP](https://pkg.go.dev/net#ParseIP) method, as that also supports ipv6 ip addresses.

abuchanan-airbyte replied:
Nice, great idea

abuchanan-airbyte commented:
@colesnodgrass I added more validation. I noted the error messages in the logs actually provide a regex pattern, which covers more ground than just looking for IP addresses.

The tests I added are examples that come from error logs in the telemetry.
